shared/utils/flamingo-utils.hpp

```diff
@@ -0,0 +1,99 @@
+#include <iterator>
+#include <optional>
+#include <string_view>
+#include "flamingo/shared/hook-data.hpp"
+#include "flamingo/shared/hook-installation-result.hpp"
+#include "flamingo/shared/installer.hpp"
+#include "flamingo/shared/target-data.hpp"
+
+namespace bs_hook {
+struct FlamingoHandleHelper {
+    FlamingoHandleHelper() = default;
+    ~FlamingoHandleHelper() = default;
+
+    // disable copy
+    FlamingoHandleHelper(FlamingoHandleHelper const&) = delete;
+    FlamingoHandleHelper& operator=(FlamingoHandleHelper const&) = delete;
+
+    // disable move
+    FlamingoHandleHelper(FlamingoHandleHelper&&) = delete;
+    FlamingoHandleHelper& operator=(FlamingoHandleHelper&&) = delete;
+
+    std::optional<flamingo::HookHandle> handle;
```

Comment:
Instead of holding the handle, I'd hold things to make the handle so we can make the handle just once with a .install() call and make it then as an rvalue.

Can also potentially remake this builder to make more hooks too, theoretically.